packages/medusa/src/api/store/locales/route.ts

```diff
@@ -0,0 +1,20 @@
+import { MedusaRequest, MedusaResponse } from "@medusajs/framework/http"
+import { ContainerRegistrationKeys } from "@medusajs/framework/utils"
+
+export const GET = async (req: MedusaRequest, res: MedusaResponse) => {
+  const query = req.scope.resolve(ContainerRegistrationKeys.QUERY)
+
+  const {
+    data: [store],
+  } = await query.graph({
+    entity: "store",
+    fields: ["supported_locales.*", "supported_locales.locale.*"],
+    pagination: {
+      take: 1,
+    },
+  })
+
+  res.json({
+    locales: store?.supported_locales ?? [],
```

Comment:
thought(non-blocking): should we map it to something a bit more simple?
```
{ code: "en-US", name: "English (US)" }
```

Don't think you need anything else from the locales. Makes it easier to consume in the storefront.